components/Buttons/src/MDCFloatingButton.m

```diff
@@ -273,19 +341,168 @@ - (void)layoutSubviews {
+  if (!modeToMaximumSize) {
+    return CGSizeZero;
+  }
+
+  NSValue *sizeValue = modeToMaximumSize[@(mode)];
+  if (sizeValue) {
+    return [sizeValue CGSizeValue];
+  } else {
+    return CGSizeZero;
+  }
+}
+
+- (BOOL)updateMaximumSize {
+  CGSize newSize = [self maximumSizeForMode:self.mode];
+
+  BOOL sizeChanged = !CGSizeEqualToSize(newSize, CGSizeZero)
+      && ((newSize.width < self.maximumSize.width) || (newSize.height < self.maximumSize.height));
+  super.maximumSize = newSize;
+  if (sizeChanged) {
+    [self invalidateIntrinsicContentSize];
+  }
+  return sizeChanged;
+}
+
+- (void)setContentEdgeInsets:(UIEdgeInsets)contentEdgeInsets
+                    forShape:(MDCFloatingButtonShape)shape
+                      inMode:(MDCFloatingButtonMode)mode {
+  NSMutableDictionary *modeToContentEdgeInsets = self.shapeToModeToContentEdgeInsets[@(shape)];
+  if (!modeToContentEdgeInsets) {
+    modeToContentEdgeInsets = [@{} mutableCopy];
+    self.shapeToModeToContentEdgeInsets[@(shape)] = modeToContentEdgeInsets;
+  }
+  modeToContentEdgeInsets[@(mode)] = [NSValue valueWithUIEdgeInsets:contentEdgeInsets];
+  if (shape == _shape && mode == self.mode) {
+    [self updateShape:YES];
+  }
+}
+
+- (UIEdgeInsets)contentEdgeInsetsForMode:(MDCFloatingButtonMode)mode {
+  NSMutableDictionary *modeToContentEdgeInsets = self.shapeToModeToContentEdgeInsets[@(_shape)];
+  if (!modeToContentEdgeInsets) {
+    return UIEdgeInsetsZero;
+  }
+
+  NSValue *insetsValue = modeToContentEdgeInsets[@(mode)];
+  if (insetsValue) {
+    return [insetsValue UIEdgeInsetsValue];
+  } else {
+    return UIEdgeInsetsZero;
+  }
+}
+
+- (void)updateContentEdgeInsets {
+  super.contentEdgeInsets = [self contentEdgeInsetsForMode:self.mode];
+}
+
+- (void)setHitAreaInsets:(UIEdgeInsets)insets
+                forShape:(MDCFloatingButtonShape)shape
+                  inMode:(MDCFloatingButtonMode)mode {
+  NSMutableDictionary *modeToHitAreaInsets = self.shapeToModeToHitAreaInsets[@(shape)];
+  if (!modeToHitAreaInsets) {
+    modeToHitAreaInsets = [@{} mutableCopy];
+    self.shapeToModeToHitAreaInsets[@(shape)] = modeToHitAreaInsets;
+  }
+  modeToHitAreaInsets[@(mode)] = [NSValue valueWithUIEdgeInsets:insets];
+  if (shape == _shape && mode == self.mode) {
+    [self updateShape:NO];
+  }
+}
+
+- (UIEdgeInsets)hitAreaInsetsForMode:(MDCFloatingButtonMode)mode {
+  NSMutableDictionary *modeToHitAreaInsets = self.shapeToModeToHitAreaInsets[@(_shape)];
+  if (!modeToHitAreaInsets) {
+    return UIEdgeInsetsZero;
+  }
+
+  NSValue *insetsValue = modeToHitAreaInsets[@(mode)];
+  if (insetsValue) {
+    return [insetsValue UIEdgeInsetsValue];
+  } else {
+    return UIEdgeInsetsZero;
+  }
+}
+
+- (void)updateHitAreaInsets {
+  super.hitAreaInsets = [self hitAreaInsetsForMode:self.mode];
+}
+
+- (void)updateShape:(BOOL)forceResize {
+  BOOL needsLayout = [self updateMinimumSize];
+  needsLayout |= [self updateMaximumSize];
```

Comment:
Are you doing it this way because you don't want to call `updateMaximumSize` if `updateMinimumSize` returned `YES`? Since I don't see many |= around, I'd document what this does ("Because X, we only call updateMaximumSize if updateMinimumSize didn't change the size")

Reply:
Sorry, I probably shouldn't use BOOL as a char... 😨

Reply:
Done